examples/ex24.cpp

```diff
@@ -7,6 +7,7 @@
 //               ex24 -m ../data/beam-tet.mesh
 //               ex24 -m ../data/beam-hex.mesh -o 2 -pa
 //               ex24 -m ../data/beam-hex.mesh -o 2 -pa -p 1
+//               ex24 -m ../data/beam-hex.mesh -o 2 -pa -p 2
```

Comment:
I noticed that example 24p full assembly + cuda works fine, however; the serial version produces NaNs. I support the idea of adding the missing host reads in the code to produce correct results (even if only some of the kernels work on the GPU)

Reply:
The sample runs all work for me. Did you get NaNs using a build without cuda, for the sample run `ex24 -m ../data/beam-hex.mesh -o 2 -pa -p 2`?

Reply:
The run `` ./ex24 -m ../data/beam-hex.mesh -o 2 -pa -p 2 -d cuda`` worked. The one that  caused NaNs was `` ./ex24 -o 1 -d cuda`` -- although I realize fixing might be outside of the scope of this PR. But something good to fix in general.

Reply:
Thanks for clarifying. Now I see the issue, that `ex24 -pa -d cuda` works but `ex24 -d cuda` does not. In other words, full assembly does not work with cuda. The issues were all in `SparseMatrix::DiagScale` and are fixed now. This test is working now.

Reply:
Nice! Thanks for addressing that.